app/controllers/tasks_controller.rb

```
@@ -82,7 +82,9 @@ def update
     tasks = task.update_from_params(update_params, current_user)
     tasks.each { |t| return invalid_record_error(t) unless t.valid? }
```

Somewhat unrelated thought here, but can we just have any validations throw serializable invalid record errors? That would make it so we can get rid of this validation check and just use standard ruby exception handling.

---

Maybe, not sure if you can specify what rails validations throw?

---

I'm suggesting we replace rails validations with methods that raise exceptions instead of active record errors. This is probably a discussion for a later day, and almost certainly just inspired by my dislike for Rails' introduction of a second way to handle exceptions. 

Right now every action we take on records we persist to the database have to account for two types of failures (application and database) and those different types have their own control flows. Considers the following statement:
```ruby
  ...
begin
  task = GenericTask.create(id: existing_task_id, assigned_to: valid_user, appeal: valid_appeal)
rescue ActiveRecord::RecordNotUnique => e
  # Handle cases where we failed to create a row because of a database error.
end
  if !task.valid?
    # Handle cases where we failed application-level validations (say we forgot
    # to include an assignee).
  end
  ...
``` 

We should do our best to capture all possible database, but we will inevitably fail so we will need to account for database errors and will be stuck with this duplicated error handling. Unless we make all validations enter the ruby-standard exception flow by throwing exceptions instead of attaching errors to ActiveRecord objects.

---

I think we should just be using `create!` instead of `create` as I think that will throw an error if the validation fails.